### ISSUE TITLE
Tb115: quickFilters slow to show the assistant

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/quickFilters#215]
After migrating my (very big) production account to Thunderbird 115 I am experiencing quite a big delay (about 30 seconds) before the filter assistant is shown. I checked the error log and saw that call to get header information  about the moved email `MailUtils.getMsgHdrForMsgId()` triggers a bunch of exceptions in Thunderbird Core code:

```
Database for 2016 not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
    findMsgIdInFolder resource:///modules/MailUtils.jsm:743
    findMsgIdInFolder resource:///modules/MailUtils.jsm:749
    findMsgIdInFolder resource:///modules/MailUtils.jsm:749
    getMsgHdrForMsgId resource:///modules/MailUtils.jsm:774
    createFilterAsync chrome://quickfilters/content/qFilters-worker.js:1636
    createFilterAsync chrome://quickfolders/content/quickfolders-filterWorker.js:597
    copyList chrome://quickfolders/content/quickfolders-quickMove.js:151
    execute chrome://quickfolders/content/quickfolders-quickMove.js:208
    createFolderCallback chrome://quickfolders/content/quickfolders-interface.js:3387
    (Async: promise callback)
    onCreateInstantFolder chrome://quickfolders/content/quickfolders-interface.js:3378
    findFolderName chrome://quickfolders/content/quickfolders-interface.js:5404
    (Async: EventListener.handleEvent)
    findFolderName chrome://quickfolders/content/quickfolders-interface.js:5403
    oncommand chrome://messenger/content/messenger.xhtml:1
    _fireCommand chrome://global/content/elements/search-textbox.js:229
    (Async: setTimeout handler)
    on_input chrome://global/content/elements/search-textbox.js:197
    handleEvent chrome://global/content/customElements.js:459
Database for Drafts not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for Adressen & Tel Nummern not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for Archive not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for etc not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for ImageShack not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for IOL not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for test2 not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for Junk not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for Sent not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for Templates not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for Drafts not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for test test not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for Sent not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for Templates not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for test not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for Drafts not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for An Fainne not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for ClareMoot not accessible [MailUtils.jsm:743:17](resource:///modules/MailUtils.jsm)
Database for Sent not accessible
```

I also found out that I can avoid the error condition by repairing the folders in question. This cannot be done automatically because the exceptions are not passed up to my calling code.

**Comment by RealRaven2000** [comment]
I am considering measuring the time the call to `MailUtils.getMsgHdrForMsgId()` takes and maybe displaying a message with alert to check the error console for messages relating to folders that may need repair. If this was to be an alert or sliding notification there might want to be a way of disabling it permanently.

**Comment by RealRaven2000** [comment]
I found there is an additional startServer parameter for the getMsgHdrForMsgId() - when I filled this with the server of the target folder (the folder where the email was moved to) most of the errors did not happen (because the invalid folders were on different accounts). So this means a great increase in speed for showing the  quickFilters assistant:

[quickFilters-wx-6.2.3pre1.zip](https://github.com/RealRaven2000/quickFilters/files/12803724/quickFilters-wx-6.2.3pre1.zip)

------

To try this prerelease, download the zip file and drag it into Tb Add-ons Manager (do not extract contents)

**Comment by RealRaven2000** [comment]
Implemented in 6.3 - Published 18/10/2023
